go/pools/smartconnpool/pool.go

```diff
@@ -32,10 +32,16 @@ import (
 
 var (
 	// ErrTimeout is returned if a connection get times out.
-	ErrTimeout = vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, "resource pool timed out")
+	ErrTimeout = vterrors.New(vtrpcpb.Code_RESOURCE_EXHAUSTED, "connection pool timed out")
 
 	// ErrCtxTimeout is returned if a ctx is already expired by the time the connection pool is used
-	ErrCtxTimeout = vterrors.New(vtrpcpb.Code_DEADLINE_EXCEEDED, "resource pool context already expired")
+	ErrCtxTimeout = vterrors.New(vtrpcpb.Code_DEADLINE_EXCEEDED, "connection pool context already expired")
+
+	// ErrConnPoolClosed is returned when trying to get a connection from a closed conn pool
+	ErrConnPoolClosed = vterrors.New(vtrpcpb.Code_INTERNAL, "connection pool is closed")
+
+	// PoolCloseTimeout is how long to wait for all connections to be returned to the pool during close
+	PoolCloseTimeout = 10 * time.Second
 )
```

Comment:
This is called after we have closed all the active connections. So, a 10s timeout should be ok here. I am not sure if this needs to be user-driven.